hal_st/middlewares/ble_middleware/GattClientSt.cpp

```diff
@@ -192,12 +192,14 @@ namespace hal
         auto gattProcedureEvent = *reinterpret_cast<aci_gatt_proc_complete_event_rp0*>(vendorEvent->data);
 
         really_assert(gattProcedureEvent.Connection_Handle == connectionHandle);
```

Comment:
This line should be moved to be inside the "if".

In the case gattProcedureEvent.Error_Code is not success, a connection might have failed.